workerapi/worker.thrift

```diff
@@ -0,0 +1,38 @@
+enum Status {
+  UNKNOWN = 0
+  READY = 1
+  PENDING = 2
+  FAILURE = 3
```

Comment:
I'm not sure I follow. In that case, aborted and timeout could also be folded in? I think these are all the unique cases that we might care about upstream.